lib/sgtn-client/api/translation.rb

```diff
@@ -67,97 +69,42 @@ def self.getTranslation(component, key, locale)
+    class << self
+      def initialized?
+        @initialized ||= false
       end
-      obj && obj['data']
-    end
 
-    def self.fetch(component, locale)
-      # source locale always uses source bundles
-      return Source.getBundle(component) if LocaleUtil.is_source_locale(locale)
+      def init_translations
+        # TODO: Lock to initialize?
+        env = SgtnClient::Config.default_environment
+        mode = SgtnClient::Config.configurations[env]['bundle_mode']
+        SgtnClient.logger.debug "[Translation][init_translations]mode=#{mode}"
 
-      translation_bundle_thread = Thread.new { load(component, locale) }
-      old_source_bundle = load(component, LocaleUtil.get_source_locale)
-      source_bundle = get_cs(component, LocaleUtil.get_source_locale)
-      translation_bundle = translation_bundle_thread.value
+        if mode == 'offline'
+          extend SgtnClient::TranslationLoader::LocalBundle
+        else
+          extend SgtnClient::TranslationLoader::ServerBundle
+        end
 
-      compare_source(translation_bundle, old_source_bundle, source_bundle)
-    end
+        extend SgtnClient::TranslationLoader::SourceComparer
+        extend SgtnClient::TranslationLoader::Cache
 
-    def self.compare_source(translation_bundle, old_source_bundle, source_bundle)
-      return translation_bundle if translation_bundle.nil? || source_bundle.nil? || old_source_bundle.nil?
-
-      old_source_messages = old_source_bundle['messages']
-      translation_messages = translation_bundle['messages']
-      translation_bundle['messages'] = new_translation_messages = {}
-      source_bundle['messages'].each do |key, value|
-        translation = translation_messages[key]
-        new_translation_messages[key] = if old_source_messages[key] == value && !translation.nil?
-                                          translation
-                                        else
-                                          SgtnClient::StringUtil.new(value, LocaleUtil.get_source_locale)
-        end
+        load_translations
+        @initialized = true
       end
-      translation_bundle
-    end
 
-    none_alive = proc { |_, thread| thread.nil? || thread.alive? == false }
-    to_run = proc do |cache_key|
-      cache_item = SgtnClient::CacheUtil.get_cache(cache_key)
-      cache_item&.dig(:items).nil? || CacheUtil.is_expired(cache_item)
-    end
-    @refresh_cache_operator = SingleOperation.new(none_alive, to_run) do |cache_key, _, component, locale|
-      Thread.new do
-        cache_item = SgtnClient::CacheUtil.write_cache(cache_key, fetch(component, locale))
-        # delete thread from hash after finish
-        Thread.new { @refresh_cache_operator.remove_object(cache_key) }
-        cache_item
-      end
-    end
-    def self.refresh_cache(component, locale)
-      cache_key = SgtnClient::CacheUtil.get_cachekey(component, locale)
-      @refresh_cache_operator.operate(cache_key, component, locale)
+      def load_translations; end
```

Comment:
what does this do?

Reply:
This is to load all translations(components and locales) to cache.

Reply:
How does it load all translations if it is empty?